flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/JobVertexTaskManagersHandler.java

```diff
@@ -182,30 +188,32 @@ private static JobVertexTaskManagersInfo createJobVertexTaskManagersInfo(
                 allFinished &= state.isTerminal();
                 endTime = Math.max(endTime, vertex.getStateTimestamp(state));
 
-                counts.addIOMetrics(
-                        vertex.getCurrentExecutionAttempt(),
-                        metricFetcher,
-                        jobID.toString(),
-                        jobVertex.getJobVertexId().toString());
-                MutableIOMetrics current = new MutableIOMetrics();
-                current.addIOMetrics(
-                        vertex.getCurrentExecutionAttempt(),
-                        metricFetcher,
-                        jobID.toString(),
-                        jobVertex.getJobVertexId().toString());
-                ioMetricsInfos.add(
-                        new IOMetricsInfo(
-                                current.getNumBytesIn(),
-                                current.isNumBytesInComplete(),
-                                current.getNumBytesOut(),
-                                current.isNumBytesOutComplete(),
-                                current.getNumRecordsIn(),
-                                current.isNumRecordsInComplete(),
-                                current.getNumRecordsOut(),
-                                current.isNumRecordsOutComplete(),
-                                current.getAccumulateBackPressuredTime(),
-                                current.getAccumulateIdleTime(),
-                                current.getAccumulateBusyTime()));
+                for (AccessExecution attempt : vertex.getCurrentExecutions()) {
```

Comment:
Looks to me this is not right because the attempts of an execution vertex may locate in different task managers.

Reply:
You are right. While when I was trying to do a further refactor of this handler, I found that maybe we should use only the representing execution to calculate the aggregated task manager status, or the state can be confusing since if one attempt is CANCELED (even it's a speculative one), the state of the task manager is CANCELED. I don't think this is the expected behavior. I'd revert the changes and add a comment to describe why we choose to use the representing attempt only.

Reply:
I think it can help to display all the attempts on that TM. And then we should also include them into the aggregated IO metrics of the TM. Regarding the status shown on that tab, I think it can be an aggregation of all the **representative** attempts on that task manger, so that it can be FINISHED if the job successfully finishes.

Reply:
I think it's better. I'll refactor the method in this way.